unstructured/partition/auto.py

```diff
@@ -62,7 +63,11 @@ def partition(
     exactly_one(file=file, filename=filename, url=url)
 
     if url is not None:
-        file, filetype = file_and_type_from_url(url=url, content_type=content_type)
+        file, filetype = file_and_type_from_url(
+            url=url,
+            content_type=content_type,
+            headers=headers,
+        )
     else:
         filetype = detect_filetype(
```

Comment:
Should we emit a warning if we end up in the `else` block and `headers` is being ignored?

Reply:
Oh yeah good idea, will add that in.